public/app/features/plugins/plugin_loader.ts

```diff
@@ -74,7 +74,22 @@ export async function importPluginModule({
     return importPluginModuleInSandbox({ pluginId });
   }
 
-  return SystemJS.import(modulePath);
+  let mod = await SystemJS.import(modulePath);
+  let attempts = 0;
+  // While the module's default export is a promise, wait for it to resolve.
+  // Do we need to recurse here? It seems unlikely that a module would return a promise that resolves to another promise.
+  while (
+    !('plugin' in mod) &&
+    'default' in mod &&
+    typeof mod.default === 'object' &&
```

Comment:
Could you simplify all this logic by using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve ?

Reply:
Yeah that works nicely 👍 we still need to check for absence of `plugin` and presence of `default` but the rest of the checks can go.